lib/tests/streamlit/config_util_test.py

```diff
@@ -146,3 +155,115 @@ def test_server_option_changed(self, old, new, changed):
         self.assertEqual(
             config_util.server_option_changed(old_options, new_options), changed
         )
+
+    @patch("streamlit.config_util.click.echo")
+    def test_newlines_preserved_in_description(self, patched_echo):
+        config_options = {
+            "server.customOption": ConfigOption(
+                key="server.customOption",
+                description="""
+                    This option has multiple lines.
+                    Each line should be preserved.
+                    Even this one.
+                """,
+                default_val="default",
+                type_=str,
+            )
+        }
+
+        config_util.show_config(CONFIG_SECTION_DESCRIPTIONS, config_options)
+
+        [(args, _)] = patched_echo.call_args_list
+        # Remove the ascii escape sequences used to color terminal output.
+        output = re.compile(r"\x1b[^m]*m").sub("", args[0])
+        lines = set(output.split("\n"))
+
+        assert "# This option has multiple lines." in lines
+        assert "# Each line should be preserved." in lines
+        assert "# Even this one." in lines
+
+    @patch("streamlit.config_util.click.echo")
+    def test_omits_empty_lines_at_description_start(self, patched_echo):
+        config_options = {
+            "server.customOption": ConfigOption(
+                key="server.customOption",
+                description="""
+
+                    This option's description starts from third line.
+                    All preceding empty lines should be removed.
+                """,
+                default_val="default",
+                type_=str,
+            )
+        }
+
+        config_util.show_config(CONFIG_SECTION_DESCRIPTIONS, config_options)
+
+        [(args, _)] = patched_echo.call_args_list
+        # Remove the ascii escape sequences used to color terminal output.
+        output = re.compile(r"\x1b[^m]*m").sub("", args[0])
+        lines = set(output.split("\n"))
+
+        assert "# This option's description starts from third line." in lines
+        assert "# All preceding empty lines should be removed." in lines
```

Comment:
Hm, my concern about this test is that I don't think it tests that the preceding empty lines are removed. If that's unreasonably hard to do, then I think it'd be fine to just remove this one